src/python/grpcio/commands.py

```diff
@@ -212,50 +214,39 @@ class BuildExt(build_ext.build_ext):
     LINK_OPTIONS = {}
 
     def build_extensions(self):
+
+        def compiler_ok_with_extra_std():
+            """Test if default compiler is okay with specifying c++ version
+            when invokec in C mode. GCC is okay with this, while clang is not.
+            """
+            cc_test = subprocess.Popen(
+                ['cc', '-x', 'c', '-std=c++11', '-'],
+                stdout=subprocess.PIPE,
+                stderr=subprocess.PIPE)
+            _, cc_err = cc_test.communicate(input='int main(){return 0;}')
+            return not 'invalid argument' in Text(cc_err)
+
         # This special conditioning is here due to difference of compiler
         #   behavior in gcc and clang. The clang doesn't take --stdc++11
         #   flags but gcc does. Since the setuptools of Python only support
         #   all C or all C++ compilation, the mix of C and C++ will crash.
-        #   *By default*, the macOS use clang and Linux use gcc, that's why
-        #   the special condition here is checking platform.
-        if "darwin" in sys.platform:
```

Comment:
Since macosx is no longer launching a subprocess to use make, I think you'll also need to remove the `"darwin" in sys.platform` part of the conditional in https://github.com/lifanov/grpc/blob/1d9a18407a539f02a45400b4a0b9cb839f69fc1f/setup.py#L268-L272. I think this is the cause of the failure in https://source.cloud.google.com/results/invocations/e2579ea9-d048-4e99-af90-9767f90cb3b6/targets/grpc%2Fcore%2Fpull_request%2Fmacos%2Fgrpc_basictests_opt/log

Reply:
That makes a lot of sense! They need to be built now. Thanks!